pallets/messages/src/benchmarking.rs

```diff
@@ -22,26 +22,52 @@ fn add_message<T: Config>(schema_id: SchemaId) -> DispatchResultWithPostInfo {
 	)
 }
 
+fn offchain_message<T: Config>(schema_id: SchemaId) -> DispatchResultWithPostInfo {
+	let acc: T::AccountId = whitelisted_caller();
+	let payload: IPFSPayload = IPFSPayload::new(CID::new(Vec::from("foo")), 1_000);
+	MessagesPallet::<T>::add_ipfs_message(
+		RawOrigin::Signed(acc.clone()).into(),
+		None,
+		schema_id,
+		payload,
+	)
+}
+
 benchmarks! {
-	add {
+	add_onchain_message {
 		let n in 0 .. T::MaxMessagePayloadSizeBytes::get() - 1;
 		let m in 1 .. MESSAGES;
 		let caller: T::AccountId = whitelisted_caller();
 		let input = vec![1; n as usize];
 
 		for j in 0 .. m {
 			let sid = j % SCHEMAS;
-			assert_ok!(add_message::<T>(sid.try_into().unwrap()));
+			assert_ok!(onchain_message::<T>(sid.try_into().unwrap()));
 		}
 	}: _ (RawOrigin::Signed(caller), None, 1, input)
 
+	add_ipfs_message {
+		let n in 0 .. T::MaxMessagePayloadSizeBytes::get() - 1;
+		let m in 1 .. MESSAGES;
+		let caller: T::AccountId = whitelisted_caller();
+		let input = vec![1; n as usize];
+		let cid = CID::new(input);
+		let payload_length = 1_000;
+		let payload: IPFSPayload = IPFSPayload::new(cid, payload_length);
+
+		for j in 0 .. m {
+			let sid = (j % SCHEMAS) + 51;
+			assert_ok!(offchain_message::<T>(sid.try_into().unwrap()));
```

Comment:
nit: we ideally want to fill temporary list with a  mix of both kinds of messages to have more accurate weights